neuralop/layers/discrete_continuous_convolution.py

```diff
@@ -0,0 +1,648 @@
+    """
+
+    # check that input arrays are valid point clouds in 2D
+    assert len(grid_in) == 2
+    assert len(grid_out) == 2
+    assert grid_in.shape[0] == 2
+    assert grid_out.shape[0] == 2
+
+    n_in = grid_in.shape[-1]
+    n_out = grid_out.shape[-1]
+
+    if len(kernel_shape) == 1:
+        kernel_handle = partial(_compute_support_vals_isotropic, nr=kernel_shape[0], r_cutoff=radius_cutoff)
+    elif len(kernel_shape) == 2:
+        kernel_handle = partial(_compute_support_vals_anisotropic, nr=kernel_shape[0], nphi=kernel_shape[1], r_cutoff=radius_cutoff)
+    else:
+        raise ValueError("kernel_shape should be either one- or two-dimensional.")
+
+    grid_in = grid_in.reshape(2, 1, n_in)
+    grid_out = grid_out.reshape(2, n_out, 1)
+
+    diffs = grid_in - grid_out
+    if periodic:
+        periodic_diffs = torch.where(diffs > 0.0, diffs - 1, diffs + 1)
+        diffs = torch.where(diffs.abs() < periodic_diffs.abs(), diffs, periodic_diffs)
+
+    r = torch.sqrt(diffs[0] ** 2 + diffs[1] ** 2)
+    phi = torch.arctan2(diffs[1], diffs[0]) + torch.pi
+
+    idx, vals = kernel_handle(r, phi)
+
+    idx = idx.permute(1, 0)
+
+    if normalize:
+        vals = _normalize_convolution_tensor_2d(idx, vals, grid_in, grid_out, kernel_shape, quad_weights, transpose_normalization=transpose_normalization)
+
+    return idx, vals
+
+
+class DiscreteContinuousConv2d(DiscreteContinuousConv):
+    """
+    Discrete-continuous convolutions (DISCO) on arbitrary 2d grids as implemented for [1]. To evaluate continuous convolutions on a computer, they can be evaluated semi-discretely, where the translation operation is performed continuously, and the quadrature/projection is performed discretely on a grid [2]. They are the main building blocks for local Neural Operators [1]. Forward call expects an input of shape batch_size x in_channels x n_in.
+
+    Parameters
+    ----------
+    in_channels: int
+        input channels to DISCO convolution
+    out_channels: int
+        output channels of DISCO convolution
+    grid_in: torch.Tensor or str
+        input grid in the form of a point cloud. Can also pass a string to generate a regular (tensor) grid.
+    grid_out: torch.Tensor or str
+        output grid in the form of a point cloud. Can also pass a string to generate a regular (tensor) grid.
+    kernel_shape: Union[int, List[int]]
+        kernel shape. Expects either a signle integer for isotropic kernels or two integers for anisotropic kernels
+    n_in: Tuple[int], optional
+        number of input points
+    n_out: Tuple[int], optional
+        number of output points
+    quad_weights: torch.Tensor, optional
+        quadrature weights on the input grid
+    periodic: bool, optional
+        whether the domain is periodic
+    groups: int, optional
+        number of groups in the convolution
+    bias: bool, optional
+        whether to use a bias
+    radius_cutoff: float, optional
+        cutoff radius for the kernel
```

Comment:
Could we have a little more info for this parameter, how to choose it, what it means?